webpack/webpackPlugin.js

```diff
@@ -0,0 +1,91 @@
+const { remarkPlugins } = require('../remark');
+
+/**
+ * Vendored from `sku/config/webpack/utils`.
+ *
+ * {@link https://github.com/seek-oss/sku/blob/v10.12.0/config/webpack/utils/index.js#L10-L11}
+ */
+const SKU_WEBPACK_UTILS = {
+  SVG: /\.svg$/,
+};
+
+const ruleTestsToRemove = new Set(Object.values(SKU_WEBPACK_UTILS).map(String));
+
+const mdxRule = {
+  test: /\.mdx?$/i,
+  use: [
+    {
+      loader: 'babel-loader',
+      options: {
+        presets: ['@babel/preset-env', '@babel/preset-react'],
+      },
+    },
+    {
+      loader: '@mdx-js/loader',
+      options: {
+        remarkPlugins,
+      },
+    },
+  ],
+};
+
+const createSvgRule = (compiler) => ({
+  test: /\.svg$/i,
+  use: [
+    {
+      loader: require.resolve('file-loader'),
+      options: {
+        // Avoid emitting static assets twice, which is unnecessary and can lead
+        // to mismatches in markup.
+        emitFile: compiler.options.name === 'client',
+      },
+    },
+    {
+      loader: require.resolve('svgo-loader'),
+      options: {
+        plugins: [
+          {
+            addAttributesToSVGElement: {
+              attribute: 'focusable="false"',
+            },
+          },
+          {
+            removeViewBox: false,
+          },
+          {
+            inlineStyles: {
+              onlyMatchedOnce: false,
+            },
+          },
+        ],
+      },
+    },
+  ],
+});
+
+class ScoobieWebpackPlugin {
+  apply(compiler) {
+    /**
+     * Get rid of the SVG loaders from `SkuWebpackPlugin`.
+     *
+     * `raw-loader` loads the SVG as raw HTML, which prevents us from styling
```

Comment:
Does this mean SVGs were being represented in Remark as a massive nested HTML node instead of an image node? And it happened to work as long as you didn’t want to do anything with the image?

Reply:
Yeah, they were all inlined via `dangerouslySetInnerHTML`. We probably have to keep this behaviour around for our Mermaid diagrams to be globally styled, but hopefully we can encapsulate the Mermaiding in this codebase and have all other SVGs load as images.